lib/identity_cache/query_api.rb

```diff
@@ -139,13 +139,31 @@ def record_from_coder(coder) #:nodoc:
         end
       end
 
+      def set_inverse_of_cached_has_many(record, association_reflection, child_records)
+        associated_class = association_reflection.klass
+        return unless associated_class < IdentityCache
+
+        inverse_name = record.class.cached_has_manys.fetch(association_reflection.name).fetch(:inverse_name)
+        inverse_cached_association = associated_class.cached_belongs_tos[inverse_name]
+        return unless inverse_cached_association
+
+        prepopulate_method_name = inverse_cached_association.fetch(:prepopulate_method_name)
+        child_records.each { |child_record| child_record.send(prepopulate_method_name, record) }
```

Comment:
Oh, it looks like we define these methods as public.  However, I don't think they should be part of the public API.  They also do the same thing as `record.instance_variable_set(:"@#{options.fetch(:records_variable_name)}", cached_association)` which we do elsewhere.